calc_app/views.py

```diff
@@ -0,0 +1,149 @@
+from .models import Category, ExpenseItem
+from .forms import CategoryModelForm, ExpenseItemModelForm
+from .aggregations import get_categories_costs
+from .utils import get_plot, get_month_name, get_months_numbs_and_names
+
+from django.shortcuts import render
+from django.views.generic import View
+
+
+def get_global_context(items, month):
+    total_sum = sum([item.cost for item in items])
+    month_name = get_month_name(month)
+    months = get_months_numbs_and_names()
+    categories = Category.objects.all().order_by("name")
+    return {
+        "items": items,
+        "total_sum": total_sum,
+        "month_name": month_name,
+        "months": months,
+        "categories": categories,
+    }
+
+
+class SortByMonth(View):
+    """
+    view responsible for grouping by months
+    """
+
+    def get_context(self, month):
+        items = (
+            ExpenseItem.objects.select_related("category")
+            .filter(date__month=month)
+            .order_by("category__name")
+        )
+        return get_global_context(items, month)
+
+    def get(self, request, month):
+        context = self.get_context(month)
+        return render(request, "calc_app/month_detail.html", context)
+
+
+class SortByCategoryAndMonthView(View):
+    """
+    view responsible for grouping category's items by months
+    """
+
+    def get_context(self, month, category_id):
+        items = (
+            ExpenseItem.objects.filter(category_id=category_id)
+            .filter(date__month=month)
+            .order_by("date")
+        )
+        return get_global_context(items, month)
+
+    def get(self, request, month, category_id):
+        context = self.get_context(month, category_id)
+        return render(request, "calc_app/month_category_detail.html", context)
+
+
+class CategoryListView(View):
+    """
+    view responsible for representation of category's list and the diagram
+    """
+
+    def update_graph(self):
+        categories = Category.objects.all().order_by("name")
+        names = tuple([category.name for category in categories])
+        chart = get_plot(get_categories_costs(names), tuple(names))
+        return chart
+
+    def get_context(self):
+        form = CategoryModelForm()
+        categories = Category.objects.all().order_by("name")
+        names = tuple([category.name for category in categories])
+        chart = get_plot(get_categories_costs(names), tuple(names))
+        months = get_months_numbs_and_names()
+        month_lst1 = [(i, get_month_name(i)) for i in range(1, 5)]
```

Comment:
why do we use separate lists: lst1,lst2,lst3 ?

Reply:
If it is for representation as a lable, then I believe, that there is a way to represent it as a table.

anyway, u could use just `<some_dictionary>.items()` method to retrieve list of (key, value) tuples

to generate such list u could also use list(enumerate(<some_list_of_month>))

Reply:
Using separate lists is for displaying the list of months as a table. It is temporary feature. the role of three lists can be substituted by using only month variable already used in the code. It would be improved as fast as I will find frontend developer)

Reply:
Updated